linkerd/app/outbound/src/ingress.rs

```diff
@@ -1,45 +1,61 @@
 use crate::{http, stack_labels, tcp, trace_labels, Config, Outbound};
 use linkerd_app_core::{
     config::{ProxyConfig, ServerConfig},
-    detect, discovery_rejected, drain, errors, http_request_l5d_override_dst_name_addr,
-    http_tracing, io, profiles,
+    detect, discovery_rejected, errors, http_request_l5d_override_dst_name_addr, http_tracing, io,
+    profiles,
     proxy::api_resolve::Metadata,
     svc::{self, stack::Param},
     tls,
-    transport::{self, OrigDstAddr, Remote, ServerAddr},
+    transport::{OrigDstAddr, Remote, ServerAddr},
     AddrMatch, Conditional, Error, NameAddr,
 };
 use std::convert::TryFrom;
 use thiserror::Error;
 use tracing::{debug_span, info_span};
 
-impl Outbound<()> {
+#[derive(Clone)]
+struct AllowHttpProfile(AddrMatch);
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+struct Target {
+    dst: NameAddr,
+    version: http::Version,
+}
+
+#[derive(Clone)]
+struct TargetPerRequest(http::Accept);
+
+#[derive(Debug, Error)]
+#[error("ingress-mode routing requires a service profile")]
+struct ProfileRequired;
+
+#[derive(Debug, Error)]
+#[error("ingress-mode routing requires the l5d-dst-override header")]
+struct DstOverrideRequired;
+
+#[derive(Debug, Default, Error)]
+#[error("ingress-mode routing is HTTP-only")]
+struct IngressHttpOnly;
```

Comment:
this stuff was just all moved from the bottom of the file, right? seems good to me